### ISSUE TITLE
Revisit how users can find a learn to use the `eml$*` constructors

### ISSUE THREAD

**Issue by amoeba** [ropensci/EML#334]
[I didn't find a ticket related to this but let me know if there is one]

You can create EML with this package using three modalities and those modalities can be mixed together:

1. plain old `list`s
2. `set_*` methods
3. `eml$*` constructors

They all require you to know a decent amount of the EML standard to use and probably look things up as you work. However, sometimes I think you know what element you need but not what child elements it takes. This is where the `eml$*` constructors are useful. 

That said, they're not well-documented. `?eml` gives you a nearly blank help page and I'm not sure there's help elsewhere.

Could we make this situation better? I had a few ideas:

- Look at where the README, any R help pages, and any vignettes stand and introduce the `eml$*` constructors early and make it easy to find if you forget
- Create in a ?EML help page
- Create a help page for the `eml$*` constructor 
- Create help pages for EML elements, like ?coverage, ?creator (possibly namespaced with eml_ like ?eml_creator)

Please add in any ideas or ask questions.

**Comment by mbjones** [comment]
The discussion of auto-generated docs for `eml$` is in https://github.com/ropensci/EML/issues/252#issuecomment-442595315

**Comment by cboettig** [comment]
yup, with prototype over at https://cboettig.github.io/build.eml/reference/eml.html, but still needs more test-driving to figure out what issues arise from this approach.  (Some we've discussed in https://github.com/cboettig/build.eml/issues/2)
